core/test-api/snmp/src/main/java/org/opennms/core/test/snmp/ProxySnmpAgentConfigFactory.java

```diff
@@ -54,17 +54,25 @@ public ProxySnmpAgentConfigFactory(final InputStream config) throws MarshalExcep
 
     @Override
     public SnmpAgentConfig getAgentConfig(final InetAddress address) {
-    	final SnmpAgentConfigProxyMapper mapper = SnmpAgentConfigProxyMapper.getInstance();
-    	final SnmpAgentAddress agentAddress = mapper.getAddress(address);
-    	
-    	final String addressString = str(address);
-		if (agentAddress == null) {
-			LOG.debug("No agent address mapping found for {}!  Try adding a @JUnitSnmpAgent(host=\"{}\", resource=\"...\" entry...", addressString, addressString);
-    		return super.getAgentConfig(address);
-    		// throw new IllegalArgumentException("No agent address mapping found for " + addressString);
-    	}
+        return getAgentConfig(address, null);
+    }
+
+    @Override
+    public SnmpAgentConfig getAgentConfig(final InetAddress address, String location) {
+        final SnmpAgentConfigProxyMapper mapper = SnmpAgentConfigProxyMapper.getInstance();
```

Comment:
I used already existing code but looks like SnmpAgentConfigProxyMapper is used to store InetAddress, SnmpAgentAddress in HashMap  which will be subsequently used to retrieve corresponding agent Address